controllers/storagecluster/resourceprofile.go

```diff
@@ -0,0 +1,113 @@
+package storagecluster
+
+import (
+	"errors"
+	"time"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/v4/api/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/defaults"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	resourceProfileChangeApplyingMessage = "New resource profile is being applied"
+	resourceProfileChangeFailedMessage   = "New resource profile failed to apply, please revert to the last working resource profile"
+	resourceProfileChangeTimeout         = 20 * time.Minute
```

Comment:
How is this value determined? 20 minutes is a lot of time for pods to restart with new resource requirements. Can we decrease it?

Reply:
@travisn How much time do you think will be sufficient for a typical cluster, for rook to process all the daemon restarts?

Reply:
Timeouts are a challenge because it depends on a number of factors, such as whether images are being pulled, how big the cluster is (number of OSDs), etc. In this case of updating the resources at least we should only be updating existing daemons and not deploying new daemons where images might need to be pulled. What about a more dynamic set of checks such as these?
1. If any daemons are in pending state, consider it failed immediately since it's likely pending due to lack of resources.
2. If no daemons have been updated in some short timeout, then consider it failed. For example, keep track of the number of daemons with the new profile label. If it hasn't increased for two minutes, then consider it failed. 

The first condition should actually be sufficient to catch the cause, so we may not need any timeout checking from the second bullet.

Reply:
Have removed the timeout, And implemented another logic to check if any of the coming pods with the resource profile label are stuck in pending for more than 2 mins. PTAL
@travisn @umangachapagain